sys/shell/shell.c

```diff
@@ -92,6 +98,21 @@ static shell_command_handler_t search_commands(const shell_command_t *entry,
     return NULL;
 }
 
+#if defined(MODULE_SHELL_COMMAND_XFA)
+static shell_command_handler_t search_commands_xfa(char *command)
+{
+    unsigned n = XFA_LEN(shell_command_t*, shell_commands_xfa);
+
+    for (unsigned i = 0; i < n; i++) {
+        const volatile shell_command_t *entry = shell_commands_xfa[i];
+        if (strcmp(entry->name, command) == 0) {
+            return entry->handler;
+        }
+    }
+    return NULL;
+}
+#endif
```

Comment:
This needs to be

```suggestion
static shell_command_handler_t search_commands_xfa(char *command)
{
#if IS_USED(MODULE_SHELL_COMMAND_XFA)
    unsigned n = XFA_LEN(shell_command_t*, shell_commands_xfa);

    for (unsigned i = 0; i < n; i++) {
        const volatile shell_command_t *entry = shell_commands_xfa[i];
        if (strcmp(entry->name, command) == 0) {
            return entry->handler;
        }
    }
    return NULL;
#else
    (void)command;
#endif
}
```

Reply:
missing return in else case

Reply:
Ah, yeah, the `return NULL` needs to be outside of the `#if` block, but I see you already did that :-).